packages/frontend-plugin-api/src/wiring/createExtensionBlueprint.ts

```diff
@@ -47,14 +47,14 @@ export type CreateExtensionBlueprintOptions<
   UFactoryOutput extends ExtensionDataValue<any, any>,
   TDataRefs extends { [name in string]: AnyExtensionDataRef },
 > = {
-  kind: string;
-  namespace?: string;
+  kind: string | ((params: TParams) => string);
+  namespace?: string | ((params: TParams) => string);
   attachTo: { id: string; input: string };
   disabled?: boolean;
   inputs?: TInputs;
   output: Array<UOutput>;
   config?: {
-    schema: TConfigSchema;
+    schema: TConfigSchema | ((params: TParams) => TConfigSchema);
```

Comment:
Oh right, this could be useful too. Kinda makes me wonder if we should just make it available for all of them. Which then in turn makes me wonder if we should just declare it as one big callback instead :thinking:, I do think I like this a bit more though, but worth some experiments perhaps

Reply:
Yeah, I saw an example of this pattern which we can't currently support so thought it made sense. Can't seem to get typescript to be happy though this this callback approach, as I guess it's because typescript is trying to infer `TParams` from multiple locations. Sure there's a way but let's have a look together as to what the options are.